tests/utils.py

```diff
@@ -188,6 +203,34 @@ def __init__(self, log, server='simple_server.py',
 
 
 
+  def has_server_started(self, remaining_timeout=10):
+    """ Checks if the server has started and returns boolean value"""
+
+    start = time.time()
+    while remaining_timeout > 0:
+      # Seek is needed to move the pointer to the beginning of the file, because
+      # the subprocess could have read and/or write and thus moved the pointer.
+      self.__temp_log_file.seek(0)
+      log_message = self.__temp_log_file.read()
+
+      if len(log_message) > 0:
+        lines = log_message.splitlines()
+
+        if "bind succeeded" in lines:
+          return True
+        elif "bind failed" in lines:
+          return False
+
+      # If the server process has exited we consider this as a
+      # failed attempt to connect to the server.
+      if self.__server_process.poll():
```

Comment:
```suggestion
      if self.__server_process.poll() is not None:
```
Please review this yourself as well but I think it's more pythonic to avoid even potential issues with Truthy/Falsey return values from poll()